src/streams.c

```diff
@@ -809,29 +795,29 @@ static Obj PRINT_OR_APPEND_TO(Obj args, int append)
     /* print all the arguments, take care of strings and functions         */
     for ( i = 2;  i <= LEN_PLIST(args);  i++ ) {
         arg = ELM_LIST(args,i);
-        if ( IS_PLIST(arg) && 0 < LEN_PLIST(arg) && IsStringConv(arg) ) {
-            PrintString1(arg);
-        }
-        else if ( IS_STRING_REP(arg) ) {
-            PrintString1(arg);
-        }
-        else if ( TNUM_OBJ(arg) == T_FUNCTION ) {
-            PrintFunction( arg );
-        }
-        else {
-            memcpy( readJmpError, STATE(ReadJmpError), sizeof(syJmp_buf) );
 
-            /* if an error occurs stop printing                            */
-            TRY_IF_NO_ERROR {
-                PrintObj( arg );
+        /* if an error occurs stop printing                            */
+        memcpy( readJmpError, STATE(ReadJmpError), sizeof(syJmp_buf) );
+        TRY_IF_NO_ERROR {
```

Comment:
It would be nice if the commit message also explained the motivation for moving  `TRY_IF_NO_ERROR`. I assume it is because `PrintFunction` also can execute GAP code, and thus could trigger a `longjmp`?